File: backend/Face/src/main/java/com/eureka/face/entity/Face.java
package com.eureka.face.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity

@Table
public class Face {

    @Id
    @GeneratedValue
    private int id;
    private int product;
    private int user;
    private float happy;
    private float neutral;
    private float sad;
    private float angry;
    private float fearful;
    private float disgusted;
    private float surprised;
    private int time;
    private String createdAt;

}

File: frontend/src/api/face.js
import { request } from './index.js';

const faceDomain = 'http://k4d104.p.ssafy.io:8088/face';

export const face = {
  // 감정 분석 결과 저장
  create(angry, createdAt, disgusted, fearful, happy, id, neutral, product, sad, surprised, time, user) {
    return request(faceDomain, 'post', `/add`, { angry, createdAt, disgusted, fearful, happy, id, neutral, product, sad, surprised, time, user });
  },

  //감정 분석 결과 삭제
  delete(id) {
    return request(faceDomain, 'delete', `/deleteById${id}`);
  },

  //감정분석 결과 조회
  fetch(user) {
    return request(faceDomain, 'get', `/getAllByUser/${user}`);
  },

  fetchEmotion(userId) {
    return request(faceDomain, 'get', `/product/${userId}`);
  },
};
